### DISCUSSION TITLE
Using PnPonline to audit a site for file changes and ownership

### DISCUSSION THREAD

**Discussion by SacJones** [pnp/powershell#443] [Q&A] [answered]
I'm trying to use the pnponline module to get audit information on one of our document sites. I can connect to the site and get a listing of files and folders using get-pnpfolderitem but there are some details I can't get - author and modifiedby, those filed come up blank.

I can throw all the files into an array and some of the fields I can get data back on, file.name, file.path.identity, file,timecreated, file.timelast modified but the important ones (author and modifiedby) are becoming a mystery. If I try to pipe the variable file|fl

I get this error - "the collection has not been initialized. It has not been requested or the request has not been executed". I don't know where else to try and figure this out. I opened a case with Microsoft and they referred me to this forum.

Any help or ideas would be appreciated - thanks!

**Comment by veronicageek** [comment]
Hi @SacJones,
Have you tried with `$item.FieldValues.Author.LookupValue` (for the CreatedBy) and `$item.FieldValues.Editor.LookupValue` (for the ModifiedBy) ?

**Reply by SacJones** [reply]
I have not, but I will give that a shot - thanks.

**Reply by SacJones** [reply]
You'll have to forgive me because I'm still trying to get the hang of this module, but fieldvalues isn't a member of the object returned from get-pnpfolderitem. I've tried get-pnplistitem and I was able to see fieldvalue.author but lookupvalue comes back blank also.

**Comment by veronicageek** [comment, marked as answer]
The following is working for me:

```powershell
$allFiles = Get-PnPListItem -List "Documents" 
$results = @()

foreach($file in $allFiles){
    $results += [pscustomobject]@{
        fileName = $file["FileLeafRef"]
        CreatedBy = $file.FieldValues.Author.LookupValue
        ModifiedBy = $file.FieldValues.Editor.LookupValue
    }
}
$results
```

![image](https://user-images.githubusercontent.com/25181757/111204847-6cc40c00-85be-11eb-8944-bd70ae4335a7.png)

**Reply by SacJones** [reply]
Wow that worked! Thank you - I would have never gotten there if I had a million years. Got any thoughts on file name, path, created date and last modified date? I would take a stab at it but clearly I was going at this the wrong way.

**Reply by SacJones** [reply]
Wait I figured it out - thank you so much for helping me with this
